src/anyon/anyon.jl

```diff
@@ -374,31 +374,179 @@ end
+            if distance != 1
+                return false
+            end
+        end
+    end
+
+    return (typeof(get_gate_symbol(gate)) in native_gates)
+end
 
-        targets = get_connected_qubits(gate)
+"""
+    is_native_instruction(
+        readout::Readout,
+        connectivity::Union{LineConnectivity,LatticeConnectivity},
+        native_gates::Vector{DataType} = set_of_native_gates,
+    )::Bool
 
-        return (get_qubits_distance(targets[1], targets[2], connectivity) == 1)
-    end
+Returns `true` if the `readout` satisfies the connectivity.
 
-    return (typeof(get_gate_symbol(gate)) in set_of_native_gates)
+It does not check to determine if the `gate` is placed at the `excluded_positions` of the
+`connectivity`.
+
+# Example
+
+```jldoctest  
+julia> connectivity = LineConnectivity(3)
+LineConnectivity{3}
+1──2──3
+
+
+julia> is_native_instruction(readout(2, 2), connectivity)
+true
+
+julia> is_native_instruction(readout(4, 4), connectivity)
+false
+
+```
+"""
+function is_native_instruction(
+    readout::Readout,
+    connectivity::GeometricConnectivity,
+    native_gates::Vector{DataType} = set_of_native_gates,
+)::Bool
+
+    num_qubits = get_num_qubits(connectivity)
+    return readout.connected_qubit <= num_qubits
 end
 
-is_native_instruction(readout::Readout, connectivity::GeometricConnectivity)::Bool =
-    !any(x -> x in connectivity.excluded_positions, get_connected_qubits(readout))
+"""
+    is_native_circuit(
+        circuit::QuantumCircuit,
+        connectivity::GeometricConnectivity,
+        native_gates::Vector{DataType} = set_of_native_gates,
+    )::Tuple{Bool,String}
+
+Returns `(true, "")` if the `circuit` only contains native instructions for the
+`connectivity` and the list of possible `native_gates`. It returns
+`(false, "error_message")` otherwise. The native gates for the Anyon QPUs are used by
+default.
+
+See [`is_native_instruction`](@ref) for more details about the identification of native
+instructions.
 
+# Example
+
+```jldoctest  
+julia> connectivity = LineConnectivity(3)
+LineConnectivity{3}
+1──2──3
+
+
+julia> native_circuit = QuantumCircuit(
+           qubit_count = 3,
+           instructions = [sigma_x(1), control_z(2, 3)]
+       )
+Quantum Circuit Object:
+   qubit_count: 3 
+   bit_count: 3 
+q[1]:──X───────
+               
+q[2]:───────*──
+            |  
+q[3]:───────Z──
+               
+
+julia> is_native_circuit(native_circuit, connectivity)
+(true, "")
+
+julia> foreign_circuit = QuantumCircuit(
+           qubit_count = 3,
+           instructions = [sigma_x(1), toffoli(1, 2, 3)]
+       )
+Quantum Circuit Object:
+   qubit_count: 3 
+   bit_count: 3 
+q[1]:──X────*──
+            |  
+q[2]:───────*──
+            |  
+q[3]:───────X──
+               
+
+julia> is_native_circuit(foreign_circuit, connectivity)
+(false, "Instruction type Gate{Snowflurry.Toffoli} with targets [1, 2, 3] is not native on the connectivity")
```

Comment:
let's add that the reason is it has more than 2 connected qubits, otherwise it seems odd

Reply:
The function was actually returning `false` because the Toffoli gate is not in the list of native gates for the Anyon QPUs. I have changed the logic so all gates with more than 2 qubits now fail.